mesheryctl/pkg/utils/scripts.go

```diff
@@ -6,118 +6,168 @@ import (
+    
+    echo -e "\nCreating token secret for service account..."
+    cat <<EOF | kubectl apply -f -
+apiVersion: v1
+kind: Secret
+metadata:
+  name: "${SERVICE_ACCOUNT_NAME}-token"
+  namespace: "${NAMESPACE}"
+  annotations:
+    kubernetes.io/service-account.name: "${SERVICE_ACCOUNT_NAME}"
+type: kubernetes.io/service-account-token
+EOF
+    
+    sleep 10
+}`
+}
 
-	generateCFG := exec.Command("sh", "-c", script)
-	generateCFG.Stdout = os.Stdout
-	generateCFG.Stderr = os.Stderr
+func secretOpsTemplate() string {
+	return `
+get_secret_name_from_service_account() {
+    echo -e "\nGetting secret of service account ${SERVICE_ACCOUNT_NAME} on ${NAMESPACE}"
+    SECRET_NAME="${SERVICE_ACCOUNT_NAME}-token"
+    echo "Secret name: ${SECRET_NAME}"
+}
+
+extract_ca_crt_from_secret() {
+    echo -e -n "\nExtracting ca.crt from secret..."
+    kubectl get secret --namespace "${NAMESPACE}" "${SECRET_NAME}" -o jsonpath='{.data.ca\.crt}' | base64 --decode > "${TARGET_FOLDER}/ca.crt"
+    printf "done"
+}
+
+get_user_token_from_secret() {
+    echo -e -n "\nGetting user token from secret..."
+    USER_TOKEN=$(kubectl get secret --namespace "${NAMESPACE}" "${SECRET_NAME}" -o json | jq -r '.data["token"]' | base64 --decode)
+    if [ -z "${USER_TOKEN}" ]; then
+        echo "Error: Failed to get user token"
+        exit 1
+    fi
+    printf "done"
+}`
+}
+
+func kubeConfigTemplate() string {
+	return `
+set_kube_config_values() {
+    context=$(kubectl config current-context)
+    echo -e "\nSetting current context to: $context"
+
+    CLUSTER_NAME=$(kubectl config get-contexts "$context" | awk '{print $3}' | tail -n 1)
+    echo "Cluster name: ${CLUSTER_NAME}"
+
+    ENDPOINT=$(kubectl config view -o jsonpath="{.clusters[?(@.name == \"${CLUSTER_NAME}\")].cluster.server}")
+    echo "Endpoint: ${ENDPOINT}"
+
+    if [ -z "${ENDPOINT}" ]; then
+        echo "Error: Failed to get cluster endpoint"
+        exit 1
+    fi
+
+    echo -e "\nPreparing k8s-${SERVICE_ACCOUNT_NAME}-${NAMESPACE}-conf"
+    kubectl config set-cluster "${CLUSTER_NAME}" \
+    --kubeconfig="${KUBECFG_FILE_NAME}" \
+    --server="${ENDPOINT}" \
+    --certificate-authority="${TARGET_FOLDER}/ca.crt" \
+    --embed-certs=true
+
+    kubectl config set-credentials \
+    "${SERVICE_ACCOUNT_NAME}-${NAMESPACE}-${CLUSTER_NAME}" \
+    --kubeconfig="${KUBECFG_FILE_NAME}" \
+    --token="${USER_TOKEN}"
+
+    kubectl config set-context \
+    "${SERVICE_ACCOUNT_NAME}-${NAMESPACE}-${CLUSTER_NAME}" \
+    --kubeconfig="${KUBECFG_FILE_NAME}" \
+    --cluster="${CLUSTER_NAME}" \
+    --user="${SERVICE_ACCOUNT_NAME}-${NAMESPACE}-${CLUSTER_NAME}" \
+    --namespace="${NAMESPACE}"
+
+    kubectl config use-context "${SERVICE_ACCOUNT_NAME}-${NAMESPACE}-${CLUSTER_NAME}" \
+    --kubeconfig="${KUBECFG_FILE_NAME}"
+}`
+}
 
-	return generateCFG.Run()
+func mainOpsTemplate() string {
+	return `
+main() {
+    mkdir -p "${TARGET_FOLDER}"
```

Comment:
please retain the functionality of check_prerequisites to avoid facing any issues, if jq/base64/awk/tail commands are missing

Reply:
@hexxdump Thanks for your review! I’ve implemented the check_prerequisites function as requested to ensure the necessary commands (jq, base64, awk, and tail) are available before the script runs. Below is the screenshot showing the current implementation:

## **Screenshot**
<img width="401" alt="Screenshot 2024-11-12 at 10 47 58 PM" src="https://github.com/user-attachments/assets/1422642c-b465-439a-9b09-03086b400734">

Reply:
**@hexxdump** Thanks for your review! I’ve implemented the `check_prerequisites` function as requested to ensure the necessary commands (`jq`, `base64`, `awk`, and `tail`) are available before the script runs. Below is the screenshot showing the current implementation:

## **Screenshot**
<img width="497" alt="Screenshot 2024-11-12 at 11 01 10 PM" src="https://github.com/user-attachments/assets/d2d036d3-ac29-4827-a6c5-27f7bfdac82f">



**It is here is the updated code**
<img width="714" alt="Screenshot 2024-11-12 at 11 04 51 PM" src="https://github.com/user-attachments/assets/f862ccc6-8b1b-458c-a094-3259c6ea8fda">